source/m2mtlvserializer.cpp

```diff
@@ -164,9 +177,44 @@ bool M2MTLVSerializer::serialize_multiple_resource(const M2MResource *resource,
     return success;
 }
 
+/* See, OMA-TS-LightweightM2M-V1_0-20170208-A, Appendix C,
+ * Data Types, Integer, TLV Format 
+ * 
+ * This is easy to change to support 8, 16 or 32 interger. */
+template <typename T> void M2MTLVSerializer::set_value_int(T value, uint8_t *buffer, uint32_t &size)
+{
+    size = sizeof(value);
+
+    /* write bytes to big endian order in buffer */
+    for (uint8_t ind=0;ind<size;ind++) {
+        buffer[size-ind-1] = ((value >> (ind*8)) & 0xFF);
+    }
+}
+
+template void M2MTLVSerializer::set_value_int<int8_t>(int8_t value, uint8_t *buffer, uint32_t &size);
```

Comment:
Mainly it's the frequently-used "language + standard library extensions" in libService need to be always available. (Not sure if there are other bits in the mbed Client base that are similar?)

So just as we get memcpy and strcpy, we should also have our added bitcopy, and just as we have native pointer accesses, we should have the big-endian unaligned accesses, That sort of thing. And ns_list needs to be functional.

Some things are available kind of by accident - eg if they're a macros, or if they get inlined. Basically if the H file happens to work without the C file, there's a quiet pass.

There's a separate argument about utility of unit tests versus module tests - I feel we have too much of the former and not enough of the latter. But for the unit tests, those core utilities need to work. Stubbing them is counterproductive - they always work and cannot fail unless you crash with an invalid pointer. They have no error returns that you might want to exercise.